core/DataTable/Filter/ReplaceSummaryRowLabel.php

```diff
@@ -69,5 +69,14 @@ public function filter($table)
                 $this->filter($subTable);
             }
         }
+        
+       
+        $summaryRow = $table->getRowFromId(DataTable::ID_SUMMARY_ROW);
+        if (!empty($summaryRow)) {
+            $subTable = $summaryRow->getSubtable();
+            if ($subTable) {
+                $this->filter($subTable);
+            }
+        }
```

Comment:
Using `->getRows()` instead of `->getRowsWithoutSummaryRow()` in the iteration above should have the same effect

Reply:
yes it needs a lot more memory and time though. We refactored this a while ago as `getRows()` is quite slow etc. When possible they should be always used separate.

Reply:
Maybe we should add a comment to the `getRows()` method in datatable class, that it should be used with caution.
At the first look it does not seem to make much difference as it simply returns a bit more:
```php
    public function getRows()
    {
        if (is_null($this->summaryRow)) {
            return $this->rows;
        } else {
            return $this->rows + array(self::ID_SUMMARY_ROW => $this->summaryRow);
        }
    }
```
But for sure internally it creates a new copy of the full rows array (to attach the summary row) instead of returning a reference...